Assets/Scripts/Models/FurnitureWorkshop.cs

```diff
@@ -0,0 +1,343 @@
+        }
+        else
+        {
+            sb.AppendLine("No selected production");
+        }
+        return sb.ToString();
+    }
+
+    public void Initialize()
+    {
+        WorkshopMenuActions = new List<WorkshopContextMenu>();
+
+        // check if context menu is needed
+        if (PossibleProductions.Count > 1)
+        {
+            furnParameters.AddParameter(new Parameter(CUR_PRODUCTION_CHAIN_PARAM_NAME, null));
+            foreach (var chain in PossibleProductions)
+            {
+                string prodChainName = chain.Name;
+                WorkshopMenuActions.Add(new WorkshopContextMenu()
+                {
+                    ProductionChainName = prodChainName,
+                    Function = (furn, prod) =>
+                    {
+                        var oldProd = furn.Parameters[CUR_PRODUCTION_CHAIN_PARAM_NAME];
+                        var isNotProcessing = furn.Parameters[CUR_PROCESSED_INV_PARAM_NAME].ToInt() == 0;
+                        // if selected production really changes and nothing is being processed now
+                        if (!prod.Equals(oldProd) && isNotProcessing)
+                        {
+                            furn.CancelJobs();
+                            furn.Parameters[CUR_PRODUCTION_CHAIN_PARAM_NAME].SetValue(prod);
+                        }
+                    }
+                });
+            }
+        }
+        else
+        {
+            if (PossibleProductions.Count == 1)
+                furnParameters.AddParameter(new Parameter(CUR_PRODUCTION_CHAIN_PARAM_NAME,
+                    PossibleProductions[0].Name));
+            else
+                Debug.ULogWarning("Furniture {0} is marked as factory, but has no production chain", furniture.Name);
+        }
+
+        // add dynamic params here
+        furnParameters.AddParameter(new Parameter(CUR_PROCESSING_TIME_PARAM_NAME, 0f));
+        furnParameters.AddParameter(new Parameter(MAX_PROCESSING_TIME_PARAM_NAME, 0f));
+        furnParameters.AddParameter(new Parameter(CUR_PROCESSED_INV_PARAM_NAME, 0));
+    }
+
+    public void SetParentFurniture(Furniture furniture)
+    {
+        this.furniture = furniture;
+    }
+
+    public void Update(float deltaTime)
+    {
+        // if there is enough input, do the processing and store item to output
+        // - remove items from input
+        // - add param to reflect factory can provide output (has output inside)
+        //   - as output will be produced after time, it is possible that output spot can be ocupied meanwhile
+        // - process for specified time
+        // - if output slot is free, provide output (if not, keep output 'inside' factory)
+
+        var curSetupChainName = furnParameters[CUR_PRODUCTION_CHAIN_PARAM_NAME].ToString();
+
+        if (!string.IsNullOrEmpty(curSetupChainName))
+        {
+            FurnitureWorkshop.ProductionChain prodChain = GetProductionChainInfo(curSetupChainName);
```

Comment:
This can be -> `ProductionChain prodChain = GetProductionChainInfo(curSetupChainName);`